### PR TITLE
Add Node.js for Rumprun

### PR DESCRIPTION
**Pull request by davedoesdev**

### CODE REVIEW

File: nodejs/patches/0001-patch-no-execinfo.patch [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+-  if (size == 0) {
+-    OS::PrintError("(empty)\n");
+-  } else {
+-    bt_sprnf_addrs(&memmap, trace, size, const_cast<char*>("%a\n"),
+-                   out, sizeof(out), NULL);
+-    OS::PrintError(out);
+-  }
+-  bt_unload_memmap(&memmap);
+-  bt_release_accessor(&acc);
+-#endif  // V8_LIBC_GLIBC || V8_OS_BSD
```
Comment (anttikantee) [inline nodejs/patches/0001-patch-no-execinfo.patch]: Ah that's what you meant by "comment out _body_".

But wouldn't it be nicer to "#if 0" it?

File: nodejs/patches/0002-patch-link-unwind-cxx.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- deps/v8/tools/gyp/v8.gyp.orig	2015-09-30 22:07:23.031436986 +0100
++++ deps/v8/tools/gyp/v8.gyp	2015-10-05 09:29:23.554558069 +0100
+@@ -1566,7 +1566,7 @@
+         ['OS=="netbsd"', {
+             'link_settings': {
+               'libraries': [
+-                '-L/usr/pkg/lib -Wl,-R/usr/pkg/lib -lexecinfo',
++                '-lc++',
```
Comment (anttikantee) [inline nodejs/patches/0002-patch-link-unwind-cxx.patch]: `-lc++` is no longer required with new enough Rumprun.  I just deleted the entire "link_settings" section when I was testing.

File: nodejs/patches/0005-patch-remove-mmap-fixed.patch [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ 
+ bool VirtualMemory::UncommitRegion(void* base, size_t size) {
+-  return mmap(base,
+-              size,
+-              PROT_NONE,
+-              MAP_PRIVATE | MAP_ANON | MAP_NORESERVE | MAP_FIXED,
+-              kMmapFd,
+-              kMmapFdOffset) != MAP_FAILED;
++  return true;
+ }
```
Comment (anttikantee) [inline nodejs/patches/0005-patch-remove-mmap-fixed.patch:31]: Again, `#if 0` is better (at least if someone wants to read the code _after_ patching).

Also, iirc, `#ifndef __RUMPRUN__` should work.

File: scripts/apply-patches.sh [hunk 4]
```diff
@@ -27,5 +32,5 @@ for patch in "$@"; do
         exit 1
     fi
     echo Applying ${patch}...
-    patch -d ${TARGET} -p1 -s < ${patch} || die
+    patch -d ${TARGET} -p${PREFIX} -s < ${patch} || die
 done
```
Comment (anttikantee) [inline scripts/apply-patches.sh:45]: Can you submit this a separate PR?

**Comment by anttikantee** [comment]
Apart from apply-patch, I think the comments are "optional".

**Comment by davedoesdev** [comment]
I'll fix it up in a day or three.

**Comment by anttikantee** [comment]
I'll fix apply-patches for you, then.  But I don't know how remove that part from your pull req.

**Comment by anttikantee** [comment]
ok, use `apply-patches -p0 ./ <patchlist>`

**Comment by anttikantee** [comment]
Can you rebase your branch to squash your commits together? (all _except_ the apply-patches one)

It's probably better to include the pkgsrc patches in the package than require all users to install cvs.  Less moving parts too => better.  I can change that once after merge, depending on your available time in the next day or three.

thanks!

**Comment by davedoesdev** [comment]
@anttikantee Can you take d858c8fd24f7eca4799413541b328539c37a6e1f into master? Might make the merge easier.

**Comment by anttikantee** [comment]
Done.

(I was first trying to apply it to rumprun, and was wondering wtf is the problem with git this time ... :-/ )

**Comment by davedoesdev** [comment]
Squashed commit on PR https://github.com/rumpkernel/rumprun-packages/pull/17
